pkg/nodeconfig/nodeconfig.go

```diff
@@ -496,6 +615,98 @@ func (nc *nodeConfig) configureWICD() error {
+func generateKubeconfig(secret *core.Secret, apiServerURL string) (clientcmdv1.Config, error) {
+	// extract ca.crt and token data fields
+	caCert := secret.Data[core.ServiceAccountRootCAKey]
+	if caCert == nil {
+		return clientcmdv1.Config{}, errors.New("unable to find CA cert")
+	}
+	token := secret.Data[core.ServiceAccountTokenKey]
+	if token == nil {
+		return clientcmdv1.Config{}, errors.New("unable to find token")
+	}
+	kubeconfig := clientcmdv1.Config{
+		Clusters: []clientcmdv1.NamedCluster{{
+			Name: "local",
+			Cluster: clientcmdv1.Cluster{
+				Server:                   apiServerURL,
+				CertificateAuthorityData: caCert,
+			}},
+		},
+		AuthInfos: []clientcmdv1.NamedAuthInfo{{
+			Name: "kubelet",
+			AuthInfo: clientcmdv1.AuthInfo{
+				Token: string(token),
+			},
+		}},
+		Contexts: []clientcmdv1.NamedContext{{
+			Name: "kubelet",
+			Context: clientcmdv1.Context{
+				Cluster:  "local",
+				AuthInfo: "kubelet",
+			},
+		}},
+		CurrentContext: "kubelet",
+	}
+	return kubeconfig, nil
+}
+
+// generateKubeletConfiguration returns the configuration spec for the kubelet Windows service
+func generateKubeletConfiguration(clusterDNS string) kubeletconfig.KubeletConfiguration {
+	// default values chosen based on the config on Linux worker nodes
+	falseBool := false
+	kubeAPIQPS := int32(50)
+	return kubeletconfig.KubeletConfiguration{
+		TypeMeta: meta.TypeMeta{
+			Kind:       "KubeletConfiguration",
+			APIVersion: "kubelet.config.k8s.io/v1beta1",
+		},
+		RotateCertificates: true,
+		ServerTLSBootstrap: true,
+		Authentication: kubeletconfig.KubeletAuthentication{
+			X509: kubeletconfig.KubeletX509Authentication{
+				ClientCAFile: windows.K8sDir + KubeletClientCAFilename,
+			},
+			Anonymous: kubeletconfig.KubeletAnonymousAuthentication{
+				Enabled: &falseBool,
+			},
+		},
+		ClusterDomain:         "cluster.local",
+		ClusterDNS:            []string{clusterDNS},
+		CgroupsPerQOS:         &falseBool,
+		RuntimeRequestTimeout: meta.Duration{Duration: 10 * time.Minute},
+		MaxPods:               250,
+		KubeAPIQPS:            &kubeAPIQPS,
+		KubeAPIBurst:          100,
+		SerializeImagePulls:   &falseBool,
+		FeatureGates: map[string]bool{
+			"LegacyNodeRoleBehavior":         false,
+			"NodeDisruptionExclusion":        true,
+			"RotateKubeletServerCertificate": true,
+			"SCTPSupport":                    true,
+			"ServiceNodeExclusion":           true,
+			"SupportPodPidsLimit":            true,
+		},
+		ContainerLogMaxSize: "50Mi",
+		SystemReserved: map[string]string{
+			"cpu":               "500m",
+			"ephemeral-storage": "1Gi",
+			"memory":            "1Gi",
+		},
+	}
+}
+
+// splitWindowsPath splits a Windows file path into the directory and base file name.
+// Example: 'C:\\k\\bootstrap-kubeconfig' --> dir: 'C:\k\', fileName: 'bootstrap-kubeconfig
+func splitWindowsPath(filepath string) (dir string, fileName string) {
```

Comment:
Consider dropping the named return parameters (`dir` and `fileName`) as they are not in use.

Reply:
I think the names add useful context to the return values

Reply:
True, and you can also achieve that by improving the documentation of the function.